### DISCUSSION TITLE
Modes of Projects

### DISCUSSION THREAD

**Discussion by kavitharaju** [friendsofagape/autographa-roadmap#2] [Q&A]
I dont know if it is too soon to start a discussion... 

But I was wondering, why do we plan to have different modes to create a project, instead of "different modes within the same project"? Especially the editor and MT modes.

**Comment by joelthe1** [comment]
This is great time for discussions!

I don't think I understand your question fully. So about modes, the idea is that Autographa (Ag) has different modes. Some or all of which maybe applicable to be used with a given project. So for example, we do not expect the editor mode or MT mode to work with a audio recording project. Let me know if there is some other specific info you are looking for.

**Comment by kavitharaju** [comment]
I will try to explain it more.

What I understood seeing the wireframe demo is that, we need to decide which mode we need **when we create a project**. So, as a user, if I need to do a bible translation project, I would first create a project in MT mode and start translating tokens. After translating some tokens I decide to see the output and correct it on draft. For this, I will have to generate and download the draft from my current project and start a new project in editor mode, right?

Then again when I decide to come back and finish the rest of token translations, it will be in my first MT project, which do not have any connection with the correction work I did on the second project in editor mode. 
Wouldn't it be better if we could do both the token translation and draft editing in the same project?
(But I also fear it will be hard to implement from NLP side. Things like identifying the re-ordering that happens in editor mode via alignments and updating the translation memory, preserving all user made changes in the next draft generations etc)

So my question is, instead of choosing a mode when we create a project, if we can switch between modes **within the same project**, wouldn't it be more useful? 
This is about the modes AgMT and Editor. If Audio needs to be connected with a USFM file(either created or uploaded on the tool), we can make Audio mode also like this, allowing it as a mode within any of the Autographa Projects.

**Reply by joelthe1** [reply]
I generally agree with this and that is also my thinking of how modes should work in Ag. But like you stated, your fear is not unfounded. In some cases switching between modes seamlessly might not be straight-forward like between editor mode and MT mode. Only for those cases we would have to have restrictions.

But for other modes like editor and audio mode we intend to allow the user to switch between them seamlessly. I think this does warrant more discussion though and we need to decide which use-cases we want to support and what the boundaries are. Also, I do we need to make a more concrete notion of what a project is.

**Reply by kavitharaju** [reply]
If we divide these concepts into different **types of projects** and different **modes within each project**, would that work?
* Ag Editing Projects
	* USFM editing mode
	* Audio editing mode
* AgMT(Translation) Projects
	* Token Translation Mode
	* Alignment Mode

**Reply by joelthe1** [reply]
The way I am thinking is that a **project** is the resource which contains the data/information. The mode can be thought of as a transformations on the project (data). So the project is oblivious the existence of modes and modes can only accept certain type of projects to operate on them.
